test/ibmq/test_ibmq_logger.py

```diff
@@ -0,0 +1,177 @@
+                                 'The logger level was set to {}, but it should '
+                                 'be {}.'.format(self.logger.level, level_value))
+
+    def test_log_file(self):
+        """Test setting up a logger by specifying a file and log level."""
+        log_level_warning = ('WARNING', logging.WARNING)
+        env_vars_to_patch = {self.log_level_env: log_level_warning[0],
+                             self.log_file_env: self.log_file}
+
+        with patch.dict('os.environ', env_vars_to_patch):
+            setup_logger(self.logger)
+            self.assertEqual(self.logger.level, log_level_warning[1],
+                             'The logger level was set to {}, but it should '
+                             'be {}.'.format(self.logger.level, log_level_warning[1]))
+
+            # Assert the file handler was created.
+            self.assertTrue(self.logger.handlers,
+                            'A file handler should have been setup, but it was not.')
+            self.assertEqual(len(self.logger.handlers), 1,
+                             'Many handlers were setup {}, but it should have only '
+                             'been one.'.format(self.logger.handlers))
+
+            # Note that only messages >= `WARNING` will be logged.
+            self.logger.debug('This is a debug message that should not be logged in the file.')
+            self.logger.warning('This is a warning message that should be logged in the file.')
+            self.logger.error('This is a error message that should be logged in the file.')
+
+            # Retrieve the file handler.
+            file_handler = self.logger.handlers[0]
+            file_path = file_handler.baseFilename
+
+            # Assert the file exists.
+            self.assertTrue(os.path.exists(file_path),
+                            'The file {} does not exist.'.format(file_path))
+
+            # Assert the messages were logged.
+            with open(file_path) as file_:
+                content_as_str = file_.read()
+                substrings_to_check = {'debug message': False, 'warning message': True,
+                                       'error message': True}
+
+                # Check whether the appropriate substrings are in the file.
+                for substring, in_file in substrings_to_check.items():
+                    if in_file:
+                        self.assertIn(substring, content_as_str,
+                                      'The substring "{}" was not found in the file {}({}).'
+                                      .format(substring, self.log_file, file_path))
+                    else:
+                        self.assertNotIn(substring, content_as_str,
+                                         'The substring "{}" was found in the file {}({}).'
+                                         .format('debug message', self.log_file, file_path))
+
+            # Delete the file.
+            os.remove(file_path)
```

Comment:
you can use a temporary file instead so you don't have to worry about deleting/conflicting